src/networkWrapper/protocols/socketio.js

```diff
@@ -49,36 +43,85 @@ class SocketIO extends RTWrapper {
   }
 
   /**
-   * Registers a callback on a room. Once 1 message is received, fires the
+   * Registers a callback on an event. Once 1 message is received, fires the
    * callback and unregister it afterward.
    *
-   * @param {string} roomId
+   * @param {string} event
    * @param {function} callback
    */
-  once(roomId, callback) {
-    this.socket.once(roomId, callback);
+  addOnceListener(event, callback) {
+    return this.addListener(event, callback, true);
+  }
+
+  once(event, callback) {
+    return this.addOnceListener(event, callback);
+  }
+
+  prependOnceListener(event, callback) {
+    return this.prependListener(event, callback, true);
   }
 
   /**
-   * Registers a callback on a room.
+   * Registers a callback on an event.
    *
-   * @param {string} roomId
+   * @param {string} event
    * @param {function} callback
    */
-  on(roomId, callback) {
-    this.socket.on(roomId, callback);
+  addListener(event, callback, once = false) {
+    this._addEventWrapper(event, callback, once);
+    super.addListener(event, callback, once);
+
+    return this;
+  }
+
+  on(event, callback) {
+    return this.addListener(event, callback);
+  }
+
+  prependListener(event, callback, once = false) {
+    this._addEventWrapper(event, callback, once);
+    return this.prependListener(event, callback, once);
```

Comment:
infinite loop ?

Reply:
I replaced `super` by `this` as a last-minute change, I don't even know why :disappointed: